src/utils/middleware/rateLimiter.ts

```diff
@@ -0,0 +1,29 @@
+import { NextFunction, Request, Response } from "express";
+import {
+  RateLimiterRes,
+  RateLimiterStoreAbstract
+} from "rate-limiter-flexible";
+
+import { log } from "../../utils/logger";
+
+export const makeRateLimiterMiddleware = (
+  rateLimiter: RateLimiterStoreAbstract
+) => (req: Request, res: Response, next: NextFunction) => {
+  const rateLimiterKey = req.ip;
+  return rateLimiter
+    .consume(rateLimiterKey)
+    .then(_ => next())
+    .catch((rateLimiterRes: RateLimiterRes) => {
+      const retryAfter = Math.ceil(rateLimiterRes.msBeforeNext / 1000);
+      log.debug("rate limiter blocking = %s", JSON.stringify(rateLimiterRes));
+      res
+        .set("Retry-After", retryAfter.toString())
+        .set("X-RateLimit-Remaining", rateLimiterRes.remainingPoints.toString())
```

Comment:
We can add another header that indicates the total amount of authorized API call rate. For example `X-RateLimit-Limit`

Reply:
what do you mean exactly ? =)

Reply:
Same of [here](https://github.com/pagopa/io-backend/pull/585#discussion_r386912737)

Reply:
merged suggestion